core/services/fluxmonitorv2/integrations_test.go

```diff
@@ -0,0 +1,946 @@
+	duration time.Duration,
+	msg string,
+) {
+	select {
+	case <-submissionReceived:
+		assert.Fail(t, "flags are up, but submission was sent")
+	case <-time.After(2 * time.Second):
+	}
+}
+
+func TestFluxMonitor_Deviation(t *testing.T) {
+	fa := setupFluxAggregatorUniverse(t)
+
+	// - add oracles
+	oracleList := []common.Address{fa.nallory.From}
+	_, err := fa.aggregatorContract.ChangeOracles(fa.sergey, emptyList, oracleList, oracleList, 1, 1, 0)
+	assert.NoError(t, err, "failed to add oracles to aggregator")
+	fa.backend.Commit()
+	checkOraclesAdded(t, fa, oracleList)
+
+	// Set up chainlink app
+	app := setupApplication(t, fa, func(cfg *orm.Config) {
+		cfg.Set("DEFAULT_HTTP_TIMEOUT", "100ms")
+		cfg.Set("TRIGGER_FALLBACK_DB_POLL_INTERVAL", "1s")
+	})
+	require.NoError(t, app.StartAndConnect())
+
+	// Create mock server
+	reportPrice := int64(100)
+	mockServer := cltest.NewHTTPMockServerWithAlterableResponse(t,
+		generatePriceResponseFn(&reportPrice),
+	)
+	t.Cleanup(mockServer.Close)
+
+	// When event appears on submissionReceived, flux monitor job run is complete
+	submissionReceived := fa.WatchSubmissionReceived(t,
+		[]common.Address{fa.nallory.From},
+	)
+
+	// Create the job
+	s := `
+	type              = "fluxmonitor"
+	schemaVersion     = 1
+	name              = "integration test"
+	contractAddress   = "%s"
+	precision = 0
+	threshold = 2.0
+	absoluteThreshold = 0.0
+
+	idleTimerPeriod = "1s"
+	idleTimerDisabled = false
+
+	pollTimerPeriod = "%s"
+	pollTimerDisabled = false
+
+	observationSource = """
+	ds1 [type=http method=GET url="%s"];
+	ds1_parse [type=jsonparse path="data,result"];
+
+	ds1 -> ds1_parse
+	"""
+		`
+
+	s = fmt.Sprintf(s, fa.aggregatorContractAddress, pollTimerPeriod, mockServer.URL)
+
+	requestBody, err := json.Marshal(models.CreateJobSpecRequest{
+		TOML: string(s),
+	})
+	assert.NoError(t, err)
+
+	initialBalance := currentBalance(t, &fa).Int64()
+
+	cltest.CreateJobViaWeb2(t, app, string(requestBody))
+
+	// Initial Poll
+	receiptBlock, answer := awaitSubmission(t, submissionReceived)
+
+	assert.Equal(t, atomic.LoadInt64(&reportPrice), answer,
+		"failed to report correct price to contract")
+	checkSubmission(t,
+		answerParams{
+			fa:              &fa,
+			roundId:         1,
+			answer:          int64(100),
+			from:            fa.nallory,
+			isNewRound:      true,
+			completesAnswer: true,
+		},
+		initialBalance,
+		receiptBlock,
+	)
+
+	// Change reported price to a value outside the deviation
+	reportPrice = int64(103)
+	receiptBlock, answer = awaitSubmission(t, submissionReceived)
+
+	assert.Equal(t, atomic.LoadInt64(&reportPrice), answer,
+		"failed to report correct price to contract")
+	checkSubmission(t,
+		answerParams{
+			fa:              &fa,
+			roundId:         2,
+			answer:          int64(103),
+			from:            fa.nallory,
+			isNewRound:      true,
+			completesAnswer: true,
+		},
+		initialBalance-fee,
+		receiptBlock,
+	)
```

Comment:
I think in this test it'd be good to add some basic assertions that the pipeline runs exist and look correct

Reply:
Added an additional check to verify the pipeline run exists for the round, and verifies the value